nixops/resources/output.py

```diff
@@ -0,0 +1,112 @@
+# -*- coding: utf-8 -*-
+
+# Arbitrary JSON.
+
+import os
+import nixops.util
+import nixops.resources
+
+import tempfile
+import shutil
+import subprocess
+import hashlib
+
+
+# For typing
+from nixops.deployment import Deployment
+from xml.etree.ElementTree import Element
+from nixops.nix_expr import Function
+from typing import Optional, List, Dict, Tuple
+
+class OutputDefinition(nixops.resources.ResourceDefinition):
+    """Definition of an Output."""
+
+    @classmethod
+    def get_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    @classmethod
+    def get_resource_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    def __init__(self, xml):
+        # type: (Element) -> None
+        nixops.resources.ResourceDefinition.__init__(self, xml)
+        temp = xml.find("attrs/attr[@name='script']/string")
+        self.script = temp.get("value") if temp is not None else ""
+        temp = xml.find("attrs/attr[@name='executable']/string")
+        self.executable = temp.get("value") if temp is not None else ""
+        temp = xml.find("attrs/attr[@name='name']/string")
+        self.name = temp.get("value") if temp is not None else ""
+
+    def show_type(self):
+        # type: () -> (str)
+        return "{0}".format(self.get_type())
+
+class OutputState(nixops.resources.ResourceState):
+    """State of an output."""
+    state = nixops.util.attr_property("state", nixops.resources.ResourceState.MISSING, int)
+    executable = nixops.util.attr_property("executable",None)
+    script = nixops.util.attr_property("script",None)
+    value = nixops.util.attr_property("value",None)
+    name =  nixops.util.attr_property("name",None)
+
+    @classmethod
+    def get_type(cls):
+        # type: () -> (str)
+        return "output"
+
+    @property
+    def resource_id(self):
+        # type: () -> Optional[str]
+        if self.value is not None:
+            # Avoid printing any potential secret information
+            return "{0}-{1}".format(self.name,hashlib.sha256(self.value).hexdigest()[:32])
+        else:
+            return None
+
+    def __init__(self, depl, name, id):
```

Comment:
not sure why this override is needed ?

Reply:
I'm not sure either, but removing it causes:
```
...
File "/home/dev/nixops/nixops/resources/__init__.py", line 76, in _set_attrs
    (self.id, n, v))
AttributeError: 'OutputState' object has no attribute 'id'
```

Reply:
Hm not sure what's causing that, but pretty sure the whole `__init__` override isn't needed :-)

Reply:
Ok I looked at this and it's caused by the `name =  nixops.util.attr_property("name",None)` declaration as name is used by nixops internally for persisting the state so it got confused.
So just switch from e.g name to `outputName`

Reply:
@AmineChikhaoui That worked. Updated. Thanks!